ha/util/consul_kv_store.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import consul
+import socket
+
+class ConsulKvStore:
+    """ Represents a Consul kv Store """
+
+    def __init__(self, prefix: str, host: str="localhost", port: int=8500):
+        """
+        Consul KV store.
+
+        Args:
+            prefix (str): Consul prefix
+            host (str): consul host
+            port (str): consul port
+
+        Example:
+            ConsulKvStore("cortx/ha/system/config", host="consul.srv", port=3000)
+            ConsulKvStore("cortx/ha/system/config")
+        Here localhost:8500 is connection and cortx/ha/system/config is prefix.
```

Comment:
I think this comment needs a correction (localhost:8500)?

Reply:
Updated